internal/database/badger/settings.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package badger
+
+import (
+	"fmt"
+	"path/filepath"
+)
+
+// Settings is the database settings.
+type Settings struct {
+	// Path is the database directory path to use.
+	// It defaults to the current directory if left unset.
+	Path *string
```

Comment:
Are there any problems with using `Path: string` with the default value being `""`?

Reply:
Well `""` is a valid path (current directory) and is also the default.
I use a string pointer so we can know when the field is left unset.
I need to write a blog post sometime to explain the whole reasoning behind this, but this should explain it more or less: https://github.com/qdm12/gosettings#settings-struct

> Each component has a settings struct, where the zero value of a field should be INVALID. For example, if the value 0 is allowed for a field, then it must be an *int field. On the contrary, you could have an int field if the zero value 0 is not valid. The reasoning behind this is that you want the zero Go value to be considered as 'unset field' so that the field value can be defaulted, merged with or overridden by another settings struct.

Reply:
... never mind, `""` is not valid and crashes the badger open function, so changing this to a non-pointer string!